### DISCUSSION TITLE
part4:nothing in browser

### DISCUSSION THREAD

**Discussion by zhang14725804** [browserengineering/book#408] [Q&A] [answered]

**Comment by pavpanchekha** [comment, marked as answer]
Hi @zhang14725804, I'm not sure what the question is here. Are you saying that nothing shows up in the browser once you start Part 4? Are there any error messages? If there aren't any, most likely `draw` isn't being called or something inside is missing, since `draw` is the code that transfers data from the Skia canvas to the SDL window.

**Reply by zhang14725804** [reply]
Sincere thanks for your help，Thank you very much。I'll take a closer look to see if I'm missing anything
